### PR TITLE
New Common ID: shared image gallery

### PR DESCRIPTION
**Pull request by jiaweitao001**

### CODE REVIEW

File: resourcemanager/commonids/shared_image_gallery.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+var _ resourceids.ResourceId = SharedImageGalleryId{}
+
+// SharedImageGalleryId is a struct representing the Resource ID for a Shared Image Gallery
+type SharedImageGalleryId struct {
```
Comment (tombuildsstuff) [inline resourcemanager/commonids/shared_image_gallery.go:16]: this is specific to a DevCenter, there's already a `SharedImageGallery` which is more widely used across the Provider, so this'll want to become:

```suggestion
type DevCenterSharedImageGalleryId struct {
```

Can we update this file and the filename to account for this?

Comment (jiaweitao001) [reply resourcemanager/commonids/shared_image_gallery.go:16]: > this is specific to a DevCenter, there's already a `SharedImageGallery` which is more widely used across the Provider, so this'll want to become:
> 
> Can we update this file and the filename to account for this?

Hi @tombuildsstuff , I just realized that it's not the correct resource id I actually would like to use. I'll correct it and update the PR accordingly.

Comment (jiaweitao001) [reply resourcemanager/commonids/shared_image_gallery.go:16]: Just updated the PR, the correct resource should be `azurerm_shared_image_gallery`.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**